lib/module/src/module.rs

```diff
@@ -362,6 +382,17 @@ where
             colocated,
         })
     }
+    /// Like `declare_func_in_func` but the declared func is referred to by name.
+    pub fn declare_func_in_func_by_name(
+        &self,
+        funcname: &str,
+        in_func: &mut ir::Function,
+    ) -> Result<ir::FuncRef, ModuleError> {
+        match self.names.get(funcname) {
+            Some(&FuncOrDataId::Func(funcid)) => Ok(self.declare_func_in_func(funcid, in_func)),
+            _ => unimplemented!(),
```

Comment:
Should this return an `IncompatibleDeclaration`, or some other error? (and ditto for the other cases below)